src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/ICertificatePal.cs

```diff
@@ -10,12 +10,15 @@
 
 namespace Internal.Cryptography
 {
+    /// <summary>Provides specific implementation for X509Certificate.</summary>
     internal interface ICertificatePal : IDisposable
```

Comment:
Would this be a less noisy change if the base interface changed instead of the derived one?

Reply:
I'm not sure that answers the question.

A whole lot of this change is changing input parameters from ICertificatePal to ICertificatePal2.  if the split was `ICertificateLegacyPal` and `ICertificatePal` instead of `ICertificatePal` and `ICertificatePal2` the change seems a whole lot smaller.  (I'm flexible on the name of the "splitting this off for mscorlib-like separation" version)

Reply:
Well, I'm still not 100% convinced that we will actually need this.

Reply:
Yeah, I can split it the other way around and it definitely makes way less noise. That's what I did ~2 weeks ago for my initial quick and dirty attempt to make it all build. I couldn't come up with a good name though, but `ICertificateLegacyPal` sounds good.

@baulig, I am open to other options, but I know this one is definitely doable since I already have the full macOS CoreFX certificate stack running with it on Mono. I'm still maintaining that to run all the unit tests.